File: app/src/main/java/wxm/KeepAccount/ui/data/show/calendar/FrgCalendarContent.java
package wxm.KeepAccount.ui.data.show.calendar;

import android.os.Bundle;
import android.view.View;
import android.widget.ListView;
import android.widget.TextView;

import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import java.math.BigDecimal;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Locale;

import butterknife.BindColor;
import butterknife.BindView;
import wxm.KeepAccount.db.DBDataChangeEvent;
import wxm.KeepAccount.ui.data.show.calendar.base.SelectedDayEvent;
import wxm.androidutil.FrgUtility.FrgSupportBaseAdv;
import wxm.androidutil.util.UtilFun;
import wxm.KeepAccount.R;
import wxm.KeepAccount.define.INote;
import wxm.KeepAccount.ui.utility.AdapterNoteDetail;
import wxm.KeepAccount.ui.utility.NoteDataHelper;
import wxm.KeepAccount.ui.utility.NoteShowInfo;

/**
 * for note pad content detail
 * Created by WangXM on 2016/12/12.
 */
public class FrgCalendarContent extends FrgSupportBaseAdv {
    // for ui
    @BindView(R.id.tv_month_day)
    TextView mTVMonthDay;

    @BindView(R.id.tv_year_month)
    TextView mTVYearMonth;

    @BindView(R.id.header_day_balance)
    TextView mTVBalance;

    @BindView(R.id.lv_body)
    ListView mLVBody;

    @BindColor(R.color.darkred)
    int mCLPay;

    @BindColor(R.color.darkslategrey)
    int mCLIncome;

    // for data
    private String mSZHotDay;
    private List<INote> mLSDayContents;

    /**
     * handler for DB data change
     * @param event     for event
     */
    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onDBDataChange(DBDataChangeEvent event) {
    }

    /**
     * handler for selected day change
     * @param event     for event
     */
    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onSelectedDayChange(SelectedDayEvent event) {
        updateContent(event.getDay());
    }

    @Override
    protected int getLayoutID() {
        return R.layout.frg_calendar_content;
    }

    @Override
    protected boolean isUseEventBus() {
        return true;
    }

    @Override
    protected void initUI(Bundle bundle)    {
        mSZHotDay = null;
        mLSDayContents = null;

        loadUI(bundle);
    }

    @Override
    protected void loadUI(Bundle bundle) {
        if (UtilFun.StringIsNullOrEmpty(mSZHotDay)) {
            setVisibility(View.INVISIBLE);
            return;
        }

        String[] arr = mSZHotDay.split("-");
        if (3 != arr.length) {
            setVisibility(View.INVISIBLE);
            return;
        }

        // for header;
        setVisibility(View.VISIBLE);
        mTVMonthDay.setText(arr[2]);
        mTVYearMonth.setText(String.format(Locale.CHINA, "%s年%s月", arr[0], arr[1]));

        NoteShowInfo ni = NoteDataHelper.Companion.getInfoByDay(mSZHotDay);
        BigDecimal bb = null != ni ? ni.getBalance() : BigDecimal.ZERO;
        mTVBalance.setText(String.format(Locale.CHINA, "%s %.02f",
                0 > bb.compareTo(BigDecimal.ZERO) ? "-" : "+",
                Math.abs(bb.floatValue())));
        mTVBalance.setTextColor(0 > bb.compareTo(BigDecimal.ZERO) ? mCLPay : mCLIncome);

        // for list body
        LinkedList<HashMap<String, INote>> c_para = new LinkedList<>();
        if (!UtilFun.ListIsNullOrEmpty(mLSDayContents)) {
            for (INote ci : mLSDayContents) {
                HashMap<String, INote> hm = new HashMap<>();
                hm.put(AdapterNoteDetail.K_NODE, ci);

                c_para.add(hm);
            }
        }

        AdapterNoteDetail ap = new AdapterNoteDetail(getActivity(), c_para);
        mLVBody.setAdapter(ap);
        ap.notifyDataSetChanged();
    }

    /// PRIVATE BEGIN
    /**
     * update hot day
     * @param day   show day(example : 2017-07-06)
     */
    private void updateContent(String day) {
        mSZHotDay = day;
        mLSDayContents = NoteDataHelper.Companion.getInstance().getNotesByDay(day);

        loadUI(null);
    }

    private void setVisibility(int vis) {
        mTVMonthDay.setVisibility(vis);
        mTVYearMonth.setVisibility(vis);
        mLVBody.setVisibility(vis);
    }
    /// PRIVATE END
}

File: app/src/main/java/wxm/KeepAccount/ui/data/show/calendar/FrgCalendarContent.kt
package wxm.KeepAccount.ui.data.show.calendar

import android.os.Bundle
import android.view.View
import android.widget.ListView
import android.widget.TextView

import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

import java.math.BigDecimal
import java.util.HashMap
import java.util.LinkedList
import java.util.Locale

import kotterknife.bindView
import wxm.KeepAccount.db.DBDataChangeEvent
import wxm.KeepAccount.ui.data.show.calendar.base.SelectedDayEvent
import wxm.androidutil.ui.frg.FrgSupportBaseAdv
import wxm.androidutil.util.UtilFun
import wxm.KeepAccount.R
import wxm.KeepAccount.item.INote
import wxm.KeepAccount.ui.utility.AdapterNoteDailyDetail
import wxm.KeepAccount.ui.utility.NoteDataHelper
import wxm.androidutil.app.AppBase
import wxm.androidutil.improve.doJudge

/**
 * for note pad content detail
 * Created by WangXM on 2016/12/12.
 */
class FrgCalendarContent : FrgSupportBaseAdv() {
    // for ui
    private val mTVMonthDay: TextView by bindView(R.id.tv_month_day)
    private val mTVYearMonth: TextView by bindView(R.id.tv_year_month)
    private val mTVBalance: TextView by bindView(R.id.header_day_balance)
    private val mLVBody: ListView by bindView(R.id.lv_body)

    private val mCLPay: Int = AppBase.getColor(R.color.darkred)
    private val mCLIncome: Int = AppBase.getColor(R.color.darkslategrey)

    // for data
    private var mSZHotDay: String? = null
    private var mLSDayContents: List<INote>? = null

    /**
     * handler for DB data change
     * @param event     for event
     */
    @Suppress("UNUSED_PARAMETER", "unused")
    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onDBDataChange(event: DBDataChangeEvent) {
        mSZHotDay?.let {
            mLSDayContents = NoteDataHelper.getNotesByDay(it)
            loadUI(null)
        }
    }

    /**
     * handler for selected day change
     * @param event     for event
     */
    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onSelectedDayChange(event: SelectedDayEvent) {
        updateContent(event.day)
    }

    override fun getLayoutID(): Int {
        return R.layout.frg_calendar_content
    }

    override fun isUseEventBus(): Boolean {
        return true
    }

    override fun initUI(bundle: Bundle?) {
        mSZHotDay = null
        mLSDayContents = null

        loadUI(bundle)
    }

    override fun loadUI(bundle: Bundle?) {
        if (UtilFun.StringIsNullOrEmpty(mSZHotDay)) {
            setVisibility(View.INVISIBLE)
            return
        }

        val arr = mSZHotDay!!.split("-".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        if (3 != arr.size) {
            setVisibility(View.INVISIBLE)
            return
        }

        // for header;
        setVisibility(View.VISIBLE)
        mTVMonthDay.text = arr[2]
        mTVYearMonth.text = String.format(Locale.CHINA, "%s年%s月", arr[0], arr[1])

        val ni = NoteDataHelper.getInfoByDay(mSZHotDay!!)
        val bb = ni.balance
        mTVBalance.text = String.format(Locale.CHINA,
                "${(bb < BigDecimal.ZERO).doJudge("-", "+")} %.02f",
                Math.abs(bb.toFloat()))
        mTVBalance.setTextColor((bb < BigDecimal.ZERO).doJudge(mCLPay, mCLIncome))

        // for list body
        val cPara = LinkedList<HashMap<String, INote>>()
        if (!UtilFun.ListIsNullOrEmpty(mLSDayContents)) {
            mLSDayContents!!.forEach {
                HashMap<String, INote>().apply {
                    put(AdapterNoteDailyDetail.K_NODE, it)
                }.let { cPara.add(it) }
            }
        }

        mLVBody.adapter = AdapterNoteDailyDetail(activity!!, cPara)
    }

    /// PRIVATE BEGIN
    /**
     * update hot day
     * @param day   show day(example : 2017-07-06)
     */
    private fun updateContent(day: String) {
        mSZHotDay = day
        mLSDayContents = NoteDataHelper.getNotesByDay(day)

        loadUI(null)
    }

    private fun setVisibility(vis: Int) {
        mTVMonthDay.visibility = vis
        mTVYearMonth.visibility = vis
        mLVBody.visibility = vis
    }
    /// PRIVATE END
}
